File: Thesis/R/factorial.R
# take input from the user
num <- scan("num.txt", what=integer(), quiet=TRUE)
#num = as.integer(readline(prompt="Enter a number: "))
factorial = 1
# check is the number is negative, positive or zero
if(num < 0) {
print("Sorry, factorial does not exist for negative numbers")
} else if(num == 0) {
print("The factorial of 0 is 1")
} else {
for(i in num:1) {
factorial = factorial * i
}
print(paste("The factorial of", num ,"is",factorial))
}

File: Thesis/perl/factorial.pl
$filename = 'num.txt';

open($file, $filename) or die "Could not open file '$filename' \n";

$number = <$file>;
chomp $number;

#$number = 5; # Change the number here 

$fact = 1;

for( $i = $number; $i > 1 ; $i=$i-1 ){
    $fact = $fact*$i;
}

print "Factorial of $number is: $fact\n";
